cmake/adios2-config-common.cmake.in

```diff
@@ -8,17 +8,19 @@ endif()
 
 include(CMakeFindDependencyMacro)
 
-enable_language(C)
-enable_language(CXX)
-
 set(ADIOS2_HAVE_Fortran @ADIOS2_HAVE_Fortran@)
-if(ADIOS2_HAVE_Fortran)
-  enable_language(Fortran)
-endif()
 
 set(ADIOS2_HAVE_MPI @ADIOS2_HAVE_MPI@)
 if(ADIOS2_HAVE_MPI)
-  find_dependency(MPI)
+  # Find the MPI components for languages enabled by this application.
+  set(_adios2_mpi_components)
+  foreach(lang C CXX Fortran)
+    if(CMAKE_${lang}_COMPILER_LOADED)
+      list(APPEND _adios2_mpi_components ${lang})
+    endif()
+  endforeach()
+  find_dependency(MPI COMPONENTS ${_adios2_mpi_components})
+  unset(_adios2_mpi_components)
```

Comment:
Oops, this is the default behavior of FindMPI anyway.  I'll drop this part.

Reply:
Also we don't use the CXX interface, only the C and Fortran interfaces to MPI.

Reply:
The `CXX` component is the mpicxx wrapper's view of the C bindings.  We need that when only the CXX language is enabled in clients.  The MPI 2 C++-only interfaces are in the `MPICXX` component which we're not using.